utils/deploy/contracts.js

```diff
@@ -138,20 +138,25 @@ async function deployEnvironment(environment) {
     deployed.OnlyManager = await deployContract("compliance/OnlyManager", opts);
     deployed.RMMakeOrders = await deployContract("riskmgmt/RMMakeOrders", opts);
     deployed.CentralizedAdapter = await deployContract("exchange/adapter/CentralizedAdapter", opts);
-    deployed.OnlyManagerCompetition = await deployContract("compliance/OnlyManagerCompetition", opts, []);
+    deployed.NoComplianceCompetition = await deployContract("compliance/NoComplianceCompetition", opts, []);
+    deployed.CompetitionCompliance = await deployContract("compliance/CompetitionCompliance", opts, [accounts[0]]);
+    const complianceAddress = (environment === "kovan" ? deployed.NoComplianceCompetition.address : deployed.CompetitionCompliance.address);
     deployed.Version = await deployContract(
       "version/Version",
       opts,
       [
         pkgInfo.version, deployed.Governance.address, mlnAddr,
-        ethTokenAddress, deployed.CanonicalPriceFeed.address, deployed.OnlyManagerCompetition.address
+        ethTokenAddress, deployed.CanonicalPriceFeed.address, complianceAddress
       ],
       () => {}, true
     );
     deployed.FundRanking = await deployContract("FundRanking", opts);
     const blockchainTime = await getChainTime();
-    deployed.Competition = await deployContract("competitions/Competition", opts, [mlnAddr, chfAddress, deployed.Version.address, accounts[0], blockchainTime, blockchainTime + 864000, 22 * 10 ** 18, 10 ** 24, 1000]);
-    await deployed.Competition.instance.batchAddToWhitelist.postTransaction(opts, [10 ** 25, [accounts[0]]]);
+    deployed.Competition = await deployContract("competitions/Competition", opts, [mlnAddr, chfAddress, deployed.Version.address, accounts[0], blockchainTime, blockchainTime + 864000, 20 * 10 ** 18, 10 ** 24, 1000]);
```

Comment:
could we add after this a transfer of MLN tokens to the Competition contract ?

Reply:
Yeah but could we assume the deployer has MLN tokens?

Reply:
think we can but better to check with @travs